src/components/Constants/UseApiConstants.tsx

```diff
@@ -7,9 +9,22 @@ import {
 export const useApiConstants = ():
   | LoadConstantsQuery['constant']
   | undefined => {
-  const { data } = useLoadConstantsQuery({
+  const { data, refetch } = useLoadConstantsQuery({
     fetchPolicy: 'cache-first',
   });
 
+  const userSavedLanguage = data?.user.preferences?.locale;
+  const [localeStorageLanguage, setLocaleStorageLanguage] = useLocalStorage(
+    `user-language`,
+    '',
+  );
+  useEffect(() => {
+    // if the language in locale storage is different than the saved language, the user may have changed languages in a different browser. If so, refetch the constants.
```

Comment:
This comment is definitely true. But this effect is also responsible for refetching the language when the user changes the preference in their own browser (not just a different browser).